cmake/SetupBuildInfoAndVersionFile.cmake

```diff
@@ -0,0 +1,14 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+file(WRITE
+  "${CMAKE_BINARY_DIR}/BuildInfoAndLibraryVersions.txt"
+  "SpECTRE Version: ${SPECTRE_VERSION}\n"
+  "Git description: ${GIT_DESCRIPTION}\n"
+  "Git branch: ${GIT_BRANCH}\n"
+  "Git hash: ${GIT_HASH}\n"
```

Comment:
If you set up the file even earlier you can append the spectre version and git info in `SpectreLoadMetadata.cmake`. Up to you.